lib/spack/spack/test/env.py

```diff
@@ -995,3 +995,67 @@ def test_using_multiple_compilers_on_a_node_is_discouraged(tmp_path, mutable_con
     mpich = mpileaks["mpich"]
     assert mpich.satisfies("%gcc") and not mpich.satisfies("%clang")
     assert len(mpich.dependencies(virtuals=("c", "cxx"))) == 1
+
+
+@pytest.mark.parametrize(
+    "spack_yaml,expected,not_expected",
```

Comment:
minor: expand
```suggestion
    ["spack_yaml", "expected", "not_expected"],
```

Reply:
Why are we doing this? This is allowed, but actually no example in the entire pytest docs uses a list of strings. Wondering what we gain by doing things differently from what we do already.